src/python/pants/engine/addressable.py

```diff
@@ -281,47 +281,47 @@ def addressable(type_constraint):
     return _addressable_wrapper(AddressableDescriptor, type_constraint)
 
 
-class AddressableList(AddressableDescriptor):
+class AddressableSequence(AddressableDescriptor):
     def _checked_value(self, instance, value):
         if value is None:
             return None
 
-        if not isinstance(value, MutableSequence):
+        if not isinstance(value, (list, tuple)):
```

Comment:
How about `collections.abc.Sequence` instead?

Reply:
I originally used this and got hit by strings being Sequence :( we could use `if isinstance(Sequence) and not isinstance(str)`? This seems less clear to me but is more generic.

Reply:
Agreed. I'm fine with this until the need arises and it won't.